src/emcee/ensemble.py

```diff
@@ -164,7 +174,10 @@ def random_state(self):
         so silently.
 
         """
-        return self._random.get_state()
+        try:
+            return self._random.get_state()
+        except AttributeError:
+            return self._random.bit_generator.state
 
     @random_state.setter  # NOQA
     def random_state(self, state):
```

Comment:
Should we have:
```
try:
    if isinstance(self._random, np.random.RandomState):
         self._random.set_state(state)
    elif isinstance(self._random, Generator) and isinstance(state, dict):
        self._random.bit_generator.state = state
except:
    pass
```

Obviously `Generator` will have to be imported from somewhere, in a similar fashion to what's done in `utils.py`.

Reply:
In fact if the `state` can also be set in `_check_random_state` then you could just have:
```
try:
    self._random = _check_random_state(self._random, state)
except:
    pass
```